pandas/core/ops/__init__.py

```diff
@@ -520,13 +524,29 @@ def column_op(a, b):
     return result
 
 
-def dispatch_to_extension_op(op, left, right):
+def dispatch_to_extension_op(op, left, right, keep_null_freq: bool = False):
```

Comment:
@simonjayhawkins do we have a way of typing `left` and `right` as "not a Series or Index"?

Reply:
can u type left here (EA / np.ndarray)

Reply:
> @simonjayhawkins do we have a way of typing `left` and `right` as "not a Series or Index"?

i'm not aware of being able to exclude types.

if a particular type raises, (in this case TYpeError?) then maybe could use overloads with a return type of NoReturn https://mypy.readthedocs.io/en/latest/more_types.html#the-noreturn-type (New in version 3.5.4)

could maybe use the following pattern to allow checking with older Python...

```
if TYPE_CHECKING:
    from typing import NoReturn
else:
    NoReturn = None
```